### PR TITLE
Ido/tsk 3778 rapid7 integration http

### PR DESCRIPTION
**Pull request by IdoGafi**

### CODE REVIEW

File: plugins/rapid7/rapid7.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+func (p Rapid7Plugin) GetDefaultRequestUrl() string {
+	return ""
+}
+
+func GetNewRapid7Plugin() Rapid7Plugin {
+	return Rapid7Plugin{}
+}
+
+func (p Rapid7Plugin) GetPaginationHandler() types.PaginationHandler { //TODO: ask Oz about implementation
```
Comment (ozhey) [inline plugins/rapid7/rapid7.go:54]: Let's talk about it

File: plugins/rapid7/rapid7.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+func (p Rapid7Plugin) HandleAuth(req *http.Request, conn map[string]string) error {
+	err := connections.HandleGenericConnection(conn, req, nil, nil)
+	if err != nil {
+		return err
+	}
+	authSuffix := req.Header.Get("Authorization")
+	if authSuffix == "" {
+		errors.New("basic auth encoding failed")
+	}
```
Comment (ozhey) [inline plugins/rapid7/rapid7.go:42]: What's that?

File: plugins/rapid7/rapid7.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	return nil
+}
+
+func (p Rapid7Plugin) GetDefaultRequestUrl() string {
+	return ""
+}
+
+func GetNewRapid7Plugin() Rapid7Plugin {
```
Comment (ozhey) [inline plugins/rapid7/rapid7.go:50]: Let's change the function's name to NewPlugin, and the struct's name from Rapid7Plugin to just Plugin

The package's name is Rapyd7 so it's redundant

File: plugins/rapid7/rapid7.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	return Rapid7Plugin{}
+}
+
+func (p Rapid7Plugin) GetPaginationHandler() types.PaginationHandler { //TODO: ask Oz about implementation
+	return paging.PagePagesizePaginator{
+		PageKey:         "page",
+		PageSizeKey:     "per_page",
```
Comment (ozhey) [inline plugins/rapid7/rapid7.go:57]: This one needs to be `size`, this is the name of the parameter

File: plugins/rapid7/rapid7.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+		ResourcesKey:    "",
+		DefaultPageSize: 100,
```
Comment (ozhey) [inline plugins/rapid7/rapid7.go:59]: This should be the maximum page size Rapyd7 allows

File: plugins/rapid7/mask.yaml [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      index: 3
+      input_type: password
+      placeholder: Rapid7 Password
+      required: true
+collection:
+  name: rapid7
+  icon_uri: "https://app.dev.blinkops.com/assets/images/Icons/Companies/Company/XL/Rapid7.svg"
+  active: true
+is_connection_optional: false
+request_url_conn_type: rapid7
```
Comment (ozhey) [inline plugins/rapid7/mask.yaml:33]:
```suggestion
request_url_conn_type: rapid7
auto_paging_params:
  - page
  - size
```
You need this if you want the generator to generate the auto paging param

File: plugins/rapid7/mask.yaml [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+      body: >-
+        {
+          "email": "{{params["email"]}}",
+          "login": "{{params["login_name"]}}",
+          "name": "{{params["full_name"]}}",
+          "password": "{{params["password"]}}",
+          "passwordResetOnLogin": {{bool(params[\"password_reset_on_login\"])}},
+          "role": {
+              "id": "{{params["role_identifier"]}}",
+          }
```
Comment (ozhey) [inline plugins/rapid7/mask.yaml:98]: You have non-escaped charecters here :-(

Comment (IdoGafi) [reply plugins/rapid7/mask.yaml:98]: It's not in a function so its working, would you prefer that I'll add to make it unifed?

**Review by blink-daniel — APPROVED** [review]

**Pull request merged**